drone/deploy.go

```diff
@@ -38,16 +38,37 @@ func deploy(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
-	number, err := strconv.Atoi(c.Args().Get(1))
-	if err != nil {
-		return err
-	}
 
 	client, err := newClient(c)
 	if err != nil {
 		return err
 	}
 
+	buildArg := c.Args().Get(1)
+	var number int
+	if buildArg == "last" {
+		// Fetch the build number from the last build
+		builds, err := client.BuildList(owner, name)
+		if err != nil {
+			return err
+		}
+		for _, build := range builds {
+			if build.Status == model.StatusSuccess && build.Event != model.EventDeploy {
```

Comment:
also should consider teams that may want to filter out certain branches, to avoid deploying a feature branch. In the documentation, for example, we recommend the following code for launching jobs which takes branch into account:

```
drone build start --fork octocat/hello-world \
  $(drone build last --format="{{ .Number }}" --branch=master octocat/hello-world)
```